### ISSUE TITLE
Battery level of door/window sensor

### ISSUE THREAD

**Issue by NeoBlack** [jheling/freeathome#203]
Can you get the battery level (percent of level) for the door/window sensors?
Free@Home self has no good battery level information. Most of the time, I don't recognize that the battery is low, and then I get a message from Free@Home that the device is offline. Yeah!!! 

free@home shows the battery level and battery voltage in the device configuration:

<img width="953" alt="Bildschirmfoto 2024-09-29 um 00 14 10" src="https://github.com/user-attachments/assets/1feab311-ade1-49b1-aca6-e7fde1882d2b">

With the local API, I tried to get the value, but the response did not contain the battery data:

```json
{
  "00000000-0000-0000-0000-000000000000": {
    "devices": {
      "ABBXXXXXXXXX": {
        "deviceReboots": "21",
        "floor": "01",
        "room": "03",
        "interface": "RF",
        "deviceId": "2042",
        "displayName": "Fenster Arbeitszimmer links",
        "unresponsive": false,
        "unresponsiveCounter": 0,
        "defect": false,
        "channels": {
          "ch0001": {
            "floor": "01",
            "room": "03",
            "displayName": "Fenster Arbeitszimmer links",
            "selectedIcon": "51",
            "functionID": "64",
            "inputs": {},
            "outputs": {
              "odp0000": {
                "pairingID": 53,
                "value": "0"
              },
              "odp0001": {
                "pairingID": 41,
                "value": "0"
              }
            },
            "parameters": {}
          }
        },
        "parameters": {
          "par0065": "0"
        }
      }
    }
  }
}
```

If this integration can provide this data, I could create automation to monitor and warn before the device goes offline.

**Comment by jheling** [comment]
Hi, i looked at you initial issue with the wireless window sensor, I see a batteryVoltage attribute in the free@home dump.  Maybe we can use this?

**Comment by NeoBlack** [comment]
Yes, you are right. In the XML dump, I see the attributes `batteryStatus="FF" batteryLow="false"` and an attribute node:

```xml
<attribute name="batteryVoltage">2.98389</attribute>
```

I have no idea what the value `FF` means, and I can't find a percentage value for the battery in the XML dump. But at least two of the three values (batteryVoltage and batteryLow) are useful information.

**Comment by derjoerg** [comment]
Just two possibilities to think about (only a shot in the dark):

1. Battery Percentage might be just calculated from Battery Voltage (so e.g. 3 = 100%, 2,9 = 97%, ...)
2. The Battery Status is a hex representation of the Battery Percentage (so FF = 100%, FA = 98%)

But these are just guesses. I don't have such devices to test

**Comment by NeoBlack** [comment]
I tried to get a new dump from my door sensors, but with the new firmware, I can't find the monitor/dump view anymore.
Where can I find it in the current firmware? Then, I could check different devices and validate the hex representation theory.
